redis/src/types.rs

```diff
@@ -1310,7 +1310,7 @@ impl FromRedisValue for bool {
 impl FromRedisValue for CString {
     fn from_redis_value(v: &Value) -> RedisResult<CString> {
         match *v {
-            Value::Data(ref bytes) => Ok(CString::new(bytes.clone())?),
+            Value::Data(ref bytes) => Ok(CString::new(bytes.as_slice())?),
```

Comment:
This is also ultimately a clone, though, isn't it?

Reply:
`Vec::as_slice` doesn't allocate, but `CString::new` does allocate.
IMO this looks legit.